### PR TITLE
Add Buffer for `hex` and backslash support

### PR DESCRIPTION
**Pull request by lujiajing1126**
- [x] Add support for backslash; fix issue #12 
- [x] Add support for `hex` conversion; fix issue #17 
- [x] Add `jest` test cases

I should mention here that the treatment of hex in this program
is different from the that in the official `redis-cli` in the following manner:

We always use `UTF-8` encoding while processing `\x` unicode characters.
As mentioned in the antirez/redis#3803, which is also created by @stockholmux ,
the official `redis-cli` uses `linenoise` library which converts `\x` characters into 
`UTF-16` encoding binary causing the result described in #17 .

Pls help review the code and discuss this feature further.
@gkorland @stockholmux

### CODE REVIEW

File: lib/redis.js [hunk 1]
```diff
@@ -72,16 +73,17 @@ class RedisClient {
 	}
 
 	attachEvent() {
-		const rl = readline.createInterface(process.stdin, process.stdout);
+		this.rl = readline.createInterface(process.stdin, process.stdout);
```
Comment (gkorland) [inline lib/redis.js]: @lujiajing1126 why do yo need rl on this?

File: lib/buf.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+const BACK_SLASH = 0x5C;
+
+class InputBuffer {
+    constructor(input) {
+        this._buf = input instanceof Buffer ? input : Buffer.from(input);
+        this._replace();
+    }
+
+    _replace() {
+        if (this._buf.indexOf(BACK_SLASH) < 0) {
```
Comment (gkorland) [inline lib/buf.js]: replace indexOf<0 with !includes

File: lib/buf.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+const BACK_SLASH = 0x5C;
+
+class InputBuffer {
+    constructor(input) {
+        this._buf = input instanceof Buffer ? input : Buffer.from(input);
```
Comment (gkorland) [inline lib/buf.js]: when can input be a Buffer?

File: lib/redis.js (hunk 1 above)
Comment (lujiajing1126) [reply lib/redis.js]: Previously, I want to use some other `print` methods...
I could revert this change.

File: lib/buf.js (hunk 2 above)
Comment (lujiajing1126) [reply lib/buf.js]: I cannot understand the change here?
I understand, for the character, these two methods are the same.
As shown in this link:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes#Polyfill

File: lib/buf.js (hunk 3 above)
Comment (lujiajing1126) [reply lib/buf.js]: For now, just for some rigid test cases.

File: lib/buf.js (hunk 2 above)
Comment (gkorland) [reply lib/buf.js]: You're right they are the same, but I think includes() is more readable since this is what you really want to achieve

Comment (lujiajing1126) [reply lib/buf.js]: Yes, you are right. It's more semantic.
Changed.

**Comment by lujiajing1126** [comment]
And, for now, we run the tests with three latest node LTS versions:
`node-v6`, `node-v8` and `node-v10`

**Pull request merged**